test/Garnet.test/LuaScriptRunnerTests.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using Garnet.common;
+using Garnet.server;
+using NLua.Exceptions;
+using NUnit.Framework;
+
+namespace Garnet.test
+{
+    [TestFixture]
+    internal class LuaScriptRunnerTests
+    {
+        [Test]
+        public void CannotRunUnsafeScript()
+        {
+            // Try to load an assembly
+            using (var runner = new LuaRunner("luanet.load_assembly('mscorlib')"))
+            {
+                runner.Compile();
+                var ex = Assert.Throws<LuaScriptException>(() => runner.Run());
+                Assert.AreEqual("[string \"luanet.load_assembly('mscorlib')\"]:1: attempt to index a nil value (global 'luanet')", ex.Message);
+            }
+
+            // Try to call a OS function
+            using (var runner = new LuaRunner("os = require('os'); return os.time();"))
+            {
+                runner.Compile();
+                var ex = Assert.Throws<LuaScriptException>(() => runner.Run());
+                Assert.AreEqual("[string \"os = require('os'); return os.time();\"]:1: attempt to call a nil value (global 'require')", ex.Message);
+            }
+
+            // Try to execute the input stream
+            using (var runner = new LuaRunner("dofile();"))
+            {
+                runner.Compile();
+                var ex = Assert.Throws<LuaScriptException>(() => runner.Run());
+                Assert.AreEqual("[string \"dofile();\"]:1: attempt to call a nil value (global 'dofile')", ex.Message);
+            }
+
+            // Try to call a windows executable
+            using (var runner = new LuaRunner("require \"notepad\""))
+            {
+                runner.Compile();
+                var ex = Assert.Throws<LuaScriptException>(() => runner.Run());
+                Assert.AreEqual("[string \"require \"notepad\"\"]:1: attempt to call a nil value (global 'require')", ex.Message);
```

Comment:
does this test-case make sense when running on linux?

Reply:
it doesn't hurt so there is no strong need to special case it.